Assets/Plugins/Mediabox/Source/GameKit/GameManager/PauseHandler.cs

```diff
@@ -2,30 +2,58 @@
 
 namespace Mediabox.GameKit.GameManager {
     public class PauseHandler {
-        float timeScaleBeforePause;
-        float volumeBeforePause;
-        bool isPaused;
+        float timeScaleBeforePause = 1;
+        float volumeBeforePause = 1;
+        bool isPaused = false;
+
+        public delegate void OnPause(bool pausing);
+        public event OnPause onPause;
```

Comment:
I actually think this change is much better than having the pause handler at all. As we currently have to do the work of locating the PauseHandler before we can subscribe to its delegate event to do the unique pausing functions we want. Putting the default pause into the Game base, and then letting us override it I think will be much simpler for new users to the plugin to work with.

Reply:
The Game base float value for timeScaleBeforePause, will have to be initialized as 1. As the Game will unload when exiting, and a new Game will load when entering, and then that game will not have a stored value to react to the UnPause. This might be a nicer pattern, as we don't have to call Reset values at all, as each new game, when initialized will be providing reset values.